cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -66,4 +68,43 @@ public Executable preCompile(SlangSource source) {
 
         return executable;
     }
+
+    @Override
+    public Set<SystemProperty> loadSystemProperties(SlangSource source) {
+        try {
+            ParsedSlang parsedSlang = parseSource(source);
+            return extractProperties(parsedSlang);
+        } catch (Throwable ex) {
+            throw new RuntimeException(
+                    "Error loading properties file: " + source.getName() + " nested exception is " + ex.getMessage(),
+                    ex
+            );
+        }
+    }
+
+    private ParsedSlang parseSource(SlangSource source) {
```

Comment:
This method is not for parsing a general Slang source, it's only for parsing system properties source file. So we should change the name to describe what it does